langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/TypeParamTest.java

```diff
@@ -40,8 +40,6 @@ public void testTypeParamNegative() {
         CompileResult result = BCompileUtil.compile("test-src/type-param/type_param_test_negative.bal");
         int err = 0;
         BAssertUtil.validateError(result, err++, "incompatible types: expected 'boolean[]', found 'int[]'", 21, 20);
-        BAssertUtil.validateError(result, err++, "incompatible types: expected 'anydata', found 'function (string) " +
```

Comment:
This too is a valid error?

Reply:
`readonly` support "function" type right. `readonly` being part of `Cloneable` shouldn't we support it now.